### ISSUE TITLE
[EX-69] Add all TODOs in a codebase

### ISSUE THREAD

**Issue by taskwarrior** [GothenburgBitFactory/taskwarrior#63]
_Tom Alcorn on 2016-03-14T16:34:46Z says:_ 

 This is a large feature request for some command like
{code}
task watch </path/to/codebase/root/>
{code}
which would add any TODOs found in files under the given path to taskwarrior. It should also automatically re-run on any change to the contents of the directory, similar to other 'watch' style commands.

Filters should work as usual, e.g.
{code}
task watch project:MyApp </path/to/my/app>
{code}
would add all the TODOs found to the MyApp project.

TODOs within the codebase might look like
{code:title=app.js}
37: // TODO: This is a hack, find a more robust solution.
{code}
which could result in a task like
{code}
task add project:MyApp "[app.js:37] This is a hack, find a more robust solution."
{code}

Task attributes could be embedded in parentheses after the TODO tag e.g.
{code}
// TODO(due:2016-03-14 +qa): Write tests for this URL parser.
{code}

There is a question of how to deal with shared codebases where some TODOs apply to other people. I propose adding a who filter so that one could write
{code}
// TODO(who:alcorn): Buy bananas.
{code}
and then the corresponding watch command would be
{code}
task watch who:alcorn </some/path/>
{code}

**Comment by taskwarrior** [comment]
Migrated metadata:
```
Created: 2016-03-14T16:34:46Z
Modified: 2017-01-16T21:20:55Z
```

**Comment by taskwarrior** [comment]
_Paul Beckingham on 2016-03-14T21:47:20Z says:_ 

 Neat.  The tricky part is knowing what has and has not been ingested already.  To do this as a one-off:

```
$ cd $PROJECT
$ grep -r TODO * | while read line; do echo "task add -- '$line'"; done
```

**Comment by taskwarrior** [comment]
_Tomas Babej on 2016-03-14T21:58:01Z says:_ 

 I don't think this is something that should be in the scope of taskwarrior. However, it would make an ideal third-party add on. A basic version of this of this probably wouldn't be more than couple of lines in bash, combining git grep and task in a clever manner. It could be then ran periodically using cron.

I hacked together a similiar project, taskwiki, which syncs taskwarrior tasks into freeform wiki files and back. I advise you to think carefully about possible modifications of the tasks in the source code and how those modifications would be mapped back. I came to the conclusion there is no universal way other than keeping a uuid note along with the textual representation of the task in the file.
